app/models/task.py

```diff
@@ -1,6 +1,50 @@
 from flask import current_app
+from requests.models import parse_header_links
 from app import db
+import requests
 
 
 class Task(db.Model):
-    task_id = db.Column(db.Integer, primary_key=True)
+    task_id = db.Column(db.Integer, autoincrement=True, primary_key=True)
+    title = db.Column(db.String)
+    description = db.Column(db.String)
+    completed_at = db.Column(db.DateTime, nullable = True)
+    goal_id = db.Column(db.Integer,db.ForeignKey('goal.goal_id'),nullable = True)
+
+    def serialize(self):
+        result = {
+            'id': self.task_id,
+            'title': self.title,
+            'description': self.description,
+            'is_complete': self.completed_at != None
+        } 
+        return result     
+
+    # def serialize_two(self):
```

Comment:
You're on the right track with this alternative helper instance method! What we need is to use `serialize` when `goal_id is None` (the task doesn't belong to a goal), and `serialize_two` when goal_id is a truthy value (the task does belong to a goal)